lib/features/modeling/behavior/TextAnnotationBehavior.js

```diff
@@ -12,6 +12,19 @@ export default function TextAnnotationBehavior(eventBus) {
 
   CommandInterceptor.call(this, eventBus);
 
+  // On Append, TextAnnotations will be created on the Root.
+  // The default for connections will create the connection in the parent of
+  // the source element, so we overwrite the parent here.
+  this.preExecute('connection.create', function(context) {
```

Comment:
An alternative here would be to fix this during modeling interaction (hover), cf. https://github.com/bpmn-io/bpmn-js/blob/aa158e2d5a7de86b8a388945bbb56f217922f1ae/lib/features/modeling/behavior/FixHoverBehavior.js#L24. 

This is where lane drop targets are currently adjusted to the containing participant: 

![capture A8sibz_optimized](https://github.com/bpmn-io/bpmn-js/assets/58601/75d8f471-58c3-4cee-bfcb-94fef9487702)

Reply:
This is not only relevant during hover/manual connecting, but also when you append a text annotation via the context pad

Reply:
Then this seems to be the sensible solution then. I'd still consider the hover fix to indicate where the thing will be dropped (no auto-append).

Reply:
I will look into this

Reply:
Adjusted hover behavior in [7a43c0e](https://github.com/bpmn-io/bpmn-js/pull/2078/commits/7a43c0e52247549ccf3697553c7a9f3f17b5885c)
![Recording 2024-01-12 at 10 16 42](https://github.com/bpmn-io/bpmn-js/assets/21984219/d10f76d4-0da9-4943-8857-d6352b7937b4)